Draft-Accepted/RFCXXXX-Command-Shell-Integration.md

```diff
@@ -0,0 +1,326 @@
+    "executable": "az",
+    "version": "1.0.0",
+    "tabCompletion": {
+        "command": {
+            "executable": "az",
+            "arguments": [
+                "completion",
+                "powershell",
+                "--commandLine",
+                "{commandLine}",
+                "--cursorPosition",
+                "{cursorPosition}"
+            ]
+        }
+    }
+}
+```
+
+or a shell specific script:
+
+```json
+{
+    "$schema": "https://schemas.microsoft.com/commandshell/2023/05/01/command.shellintegration.json",
+    "executable": "az",
+    "version": "1.0.0",
+    "tabCompletion": {
+        "script": {
+            "powershell": "az-completion.ps1",
+            "bash": "az-completion.sh",
+            "zsh": "az-completion.zsh"
+        }
+    }
+}
+```
+
+Multiple shells can be specified and up to the shell to determine which one to use.
+Both a `command` and `script` can be specified, but only one of each and also up to the shell to determine which one to use.
+
+In this PowerShell example, the `az-completion.ps1` would be called one time to register the argument completer.
+A different shell may decide to call the script for each tab completion request.
+
+### Help Registration
+
+The same manifest may define a path to the help content:
+
+```json
+{
+    "$schema": "https://schemas.microsoft.com/commandshell/2023/05/01/command.shellintegration.json",
+    "executable": "az",
+    "version": "1.0.0",
+    "help": {
+        "tags": [
+            "azure",
+            "az"
+        ],
+        "path": "./help/az-help.md"
+    }
+}
+```
+
+Optional `tags` can be specified to help with searching for help content.
+
+### Predictors and Feedback Providers Registration
+
+Specifically for PowerShell, a command may also want to optionally register a Predictor or Feedback Provider.
+
+```json
+{
+    "$schema": "https://schemas.microsoft.com/commandshell/2023/05/01/command.shellintegration.json",
+    "executable": "az",
+    "version": "1.0.0",
+    "predictor": {
+        "powershell": "az-predictor.dll"
+    },
+    "feedbackProvider": {
+        "powershell": "az-feedbackprovider.dll"
+    }
+}
+```
+
+Similar to `tabCompletion`, alternate shells can choose to adopt their own mechanism for registering these extensions
+under their shell specific key.
+
+In the case of PowerShell, predictors and feedback providers are assemblies so the value is the path to the assembly
+instead of a `.ps1` script.
+To enable support for MSIX installed tools and not load the dll directly from their location, PowerShell will
+copy the dll to the user's folder if it is not already there (SHA256 hash comparison) and load it from there:
+
+- On Unix, the dll will be copied to `$HOME/.local/share/powershell/Predictors` and `$HOME/.local/share/powershell/FeedbackProviders`
+- On Windows, the dll will be copied to `$HOME\AppData\Local\Microsoft\PowerShell\Predictors` and `$HOME\AppData\Local\Microsoft\PowerShell\FeedbackProviders`
```

Comment:
What if multiple packages all say that their `feedbackProvider` is `foo.dll`?

Reply:
That shouldn't be a problem as the caching should be in a sub-folder mapped to the original tool.  Maybe it needs to be a hash of the tool path to avoid collisions with tools that have the same name.  Also need to detect that the original version is updated and update or removed and delete.